app/src/main/java/io/gnosis/safe/ui/safe/share/ShareSafeDialog.kt

```diff
@@ -0,0 +1,106 @@
+import pm.gnosis.svalinn.common.utils.openUrl
+import pm.gnosis.svalinn.common.utils.snackbar
+import pm.gnosis.svalinn.common.utils.visible
+import timber.log.Timber
+import javax.inject.Inject
+
+class ShareSafeDialog : BaseViewBindingDialogFragment<DialogShareSafeBinding>() {
+
+    @Inject
+    lateinit var viewModel: ShareSafeViewModel
+
+    override fun inflateBinding(inflater: LayoutInflater, container: ViewGroup?): DialogShareSafeBinding =
+        DialogShareSafeBinding.inflate(layoutInflater, container, false)
+
+    override fun screenId(): ScreenId? = null
+
+    override fun inject(component: ViewComponent) {
+        component.inject(this)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        setStyle(STYLE_NO_TITLE, R.style.BaseSheetDialog)
+        super.onCreate(savedInstanceState)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        viewModel.state.observe(viewLifecycleOwner, Observer {
+            when (val viewAction = it.viewAction) {
+                is BaseStateViewModel.ViewAction.Loading -> toggleLoading(viewAction.isLoading)
+                is ShowSafeDetails -> loadSafeDetails(viewAction.safeDetails)
+                is BaseStateViewModel.ViewAction.ShowError -> showError(viewAction.error)
+            }
+        })
+        viewModel.load()
+        dismissBehaviour()
+    }
+
+    private fun dismissBehaviour() {
+        dialog?.window?.setBackgroundDrawable(ColorDrawable(0))
+        binding.root.setOnClickListener {
+            dismiss()
+        }
+        binding.cardContainer.setOnClickListener { }
+    }
+
+    private fun loadSafeDetails(safeDetails: SafeDetails) {
+        toggleLoading(false)
+        with(binding) {
+            blockies.setAddress(safeDetails.safe.address)
+            safeLocalName.text = safeDetails.safe.localName
+            safeDetails.safe.address.let { address ->
+                safeAddress.text = safeDetails.safe.address.formatEthAddress(requireContext(), addMiddleLinebreak = false)
+                safeAddress.setOnClickListener {
+                    requireContext().copyToClipboard(getString(R.string.url_copied), address.asEthereumAddressChecksumString()) {
+                        snackbar(requireView(), getString(R.string.url_copied_success))
+                    }
+                }
+                link.setOnClickListener {
+                    requireContext().openUrl(
+                        getString(
+                            R.string.etherscan_address_url,
+                            address.asEthereumAddressChecksumString()
+                        )
+                    )
+                }
+            }
+            safeQrCode.setImageBitmap(safeDetails.qrCode)
+            safeEnsName.text = safeDetails.ensName ?: getString(R.string.safe_settings_not_set)
+        }
+    }
+
+    private fun toggleLoading(loading: Boolean) {
```

Comment:
do you toggle loading for qr code generation or for ens name resolution as well?